### PR TITLE
Make mockchain UTXO serializable with serde

### PR DESCRIPTION
**Pull request by CodeSandwich**
This will allow Jormungandr to write mockchain UTXO dump to JSON returned from REST API

### CODE REVIEW

File: chain-addr/src/lib.rs [hunk 1]
```diff
@@ -74,6 +79,7 @@ impl KindType {
 
 /// An unstructured address including the
 /// discrimination and the kind of address
+#[cfg_attr(feature = "generic-serialization", derive(Serialize))]
```
Comment (NicolasDP) [inline chain-addr/src/lib.rs]: Nop, this is where it gets wrong. The address is not a generic serialisable. The serde implementation is only for the `AddressReadable` type. Not for the other types.

File: chain-impl-mockchain/Cargo.toml [hunk 2]
```diff
@@ -9,21 +9,21 @@ authors = [ "Nicolas Di Prima <nicolas.diprima@iohk.io>"
...
 
-[dependencies.cardano]
-path = "../cardano"
-features = [ "generic-serialization" ]
 
 [dev-dependencies]
+quickcheck = "0.8"
+rand = "0.6"
 chain-core = { path = "../chain-core", features=["property-test-api"]}
 chain-addr = { path = "../chain-addr", features=["property-test-api"]}
```
Comment (mzabaluev) [inline chain-impl-mockchain/Cargo.toml]: I wonder if it's bad to have these duplicated as dev-dependencies with a different feature set after this change.

Comment (CodeSandwich) [reply chain-impl-mockchain/Cargo.toml]: At the state of this commit they are needed only as dev-dependencies. Are they going to be used as normal dependencies too?

Having the same dependency in both places with different features works well, I've done this in other project.

Comment (mzabaluev) [reply chain-impl-mockchain/Cargo.toml]: > Are they going to be used as normal dependencies too?

I don't know, it's this PR that tries to list them also as normal dependencies for some reason?

Comment (CodeSandwich) [reply chain-impl-mockchain/Cargo.toml]: No, this PR moves them from normal deps to dev-deps

Comment (mzabaluev) [reply chain-impl-mockchain/Cargo.toml]: Eh, I'm talking about chain-core and chain-addr.

File: chain-impl-mockchain/Cargo.toml [hunk 3]
```diff
@@ -9,21 +9,21 @@ authors = [ "Nicolas Di Prima <nicolas.diprima@iohk.io>"
...
-chain-core = { path = "../chain-core" }
-chain-addr = { path = "../chain-addr" }
-rand = "0.6"
 num-traits = "0.2"
 num-derive = "0.2"
+serde = { version = "^1.0", optional = true }
+serde_derive = { version = "^1.0", optional = true }
+cardano = { path = "../cardano" }
+chain-core = { path = "../chain-core" }
+chain-addr = { path = "../chain-addr" }
```
Comment (mzabaluev) [inline chain-impl-mockchain/Cargo.toml]: Are chain-core and chain-addr needed as regular non-dev dependencies?

**Comment by CodeSandwich** [comment]
@mzabaluev The title seemed descriptive enough about changes themselves. Should I add their higher-level goal like usage in Jormungandr REST API?

**Comment by mzabaluev** [comment]
> @mzabaluev The title seemed descriptive enough about changes themselves. Should I add their higher-level goal like usage in Jormungandr REST API?

Yeah, it would be generally good to understand why the change is needed.

File: chain-addr/src/lib.rs (hunk 1 above)
Comment (CodeSandwich) [reply chain-addr/src/lib.rs]: I've changed `Serde` serialization of `Address` to reuse its `property::Serialize` implementation. Is it ok?

**Comment by CodeSandwich** [comment]
We should wait with merge to master for Jormungandr to be able to work with master.

**Pull request closed without merging**